app/routines/generate_reset_code.rb

```diff
@@ -0,0 +1,18 @@
+class GenerateResetCode
```

Comment:
I've also come to dislike methods that automatically save the object they are called on...

When you have those, that's an extra query you cannot avoid at times.